pkg/kapis/devops/v1alpha3/handler.go

```diff
@@ -309,7 +310,7 @@ func (h *devopsHandler) getDevOps(request *restful.Request) (devops.DevopsOperat
 
 	if kubernetesClient, err := k8s.NewKubernetesClientWithToken(token, h.k8sClient.Config().Host); err != nil {
 		return nil, err
-	} else {
+	} else if kubernetesClient != nil {
```

Comment:
The variable `kubernetesClient` will be nil while token is equal to "".

Reply:
I prefer to get an error if the token is empty. The client object should not be nil if there're no errors.

See also https://github.com/LinuxSuRen/ks-devops/blob/22926b53ff69b009693009e4c5f333761d18cc4c/pkg/client/k8s/kubernetes.go#L111

Reply:
I didn't because of `kubernetes_test` test which accepts nil token.

https://github.com/LinuxSuRen/ks-devops/blob/b51988ec443dd6b12a73a830bfaa3da5c90b6782/pkg/client/k8s/kubernetes_test.go#L21

Reply:
But I agree with you that complain an error if the token is empty.